modules/saml/lib/IdP/SAML2.php

```diff
@@ -251,6 +253,26 @@ private static function getAssertionConsumerService(
             return $firstFalse;
         }
 
+        $skipEndpointValidation = false;
+        if ($authnMessageSigned === true) {         
+            $skipEndpointValidationWhenSigned = $spMetadata->getValue('skipEndpointValidationWhenSigned', false);
+            if (is_bool($skipEndpointValidationWhenSigned) === true) {
+                $skipEndpointValidation = $skipEndpointValidationWhenSigned;
+            } elseif (is_callable($skipEndpointValidationWhenSigned) === true) {
+                $shouldSkipEndpointValidation = $skipEndpointValidationWhenSigned($spMetadata);
+                if (is_bool($shouldSkipEndpointValidation) === true) {
+                    $skipEndpointValidation = $shouldSkipEndpointValidation;
+                }
+            }
+        }
+
+        if (($AssertionConsumerServiceURL !== null) && ($skipEndpointValidation === true)) {
+            Logger::warning(
+                'Using AssertionConsumerService specified in AuthnRequest because no metadata endpoint matches and skipEndpointValidationWhenSigned was true'
+            );
```

Comment:
I'm curious why the 'warning' level for the logging. In the use cases (virtual hosting) we see for this, it would be normal and expected for the ACS url to not be in the metadata

Reply:
Agreed Patrick! Should be downgraded to info-level since it's normal operation